### DISCUSSION TITLE
Fetching data for PR build on Vercel

### DISCUSSION THREAD

**Discussion by elie222** [vercel/next.js#30097] [Help] [answered]
How do you load data in PR branches?

The code we attempted to use to make this work:

```ts
export function getBaseUrl(): string {
  if (!process.env.NEXT_PUBLIC_VERCEL_URL) return 'http://localhost:3000';

  return `https://${process.env.NEXT_PUBLIC_VERCEL_URL}`;
}

// and then:
export async function getStaticProps(context: Context) {
  await fetch(`${getBaseUrl()}/api/posts`);
}
```

But then we get an error during build.

The specific error I get when using the above approach is:

> FetchError: invalid json response body at https://vercel.com/login?next=%2Fdeployments%2Fmysite-xyz.vercel.app%3Fhost%3Dxyz.vercel.app%26redirect%3D1%26section%3D reason: Unexpected token < in JSON at position 0

Feel like this is a super common use case but I can't find any info on it.

**Comment by wwwdepot** [comment, marked as answer]
Do not use `fetch()` to call an API route in `getStaticProps`, read notes https://nextjs.org/docs/basic-features/data-fetching.

**Reply by elie222** [reply]
Woah. I've read that page multiple times and didn't notice that comment.

There are multiple fetch examples on that page: https://nextjs.org/docs/basic-features/data-fetching#example

For others looking at this, see: https://nextjs.org/docs/basic-features/data-fetching#write-server-side-code-directly
